services/api-server/src/simcore_service_api_server/services/log_streaming.py

```diff
@@ -126,10 +130,19 @@ async def log_generator(self) -> AsyncIterable[str]:
             raise LogStreamerNotRegistered(
                 f"LogStreamer for job_id={self._job_id} is not correctly registered"
             )
+        last_log_time: datetime | None = None
         while True:
             while self._queue.empty():
                 if await self._project_done():
                     return
-                await asyncio.sleep(_SLEEP_SECONDS_BEFORE_CHECK_JOB_STATUS)
+                await asyncio.sleep(
+                    0.2
+                    if last_log_time is None
+                    else min(
+                        (datetime.now(tz=timezone.utc) - last_log_time).total_seconds(),
```

Comment:
I am not sure I fully understand what you are doing here.
you wait 0.2 seconds or the time it took between the last log and this one? that is reduce the load? Then I have a question, is there no way to use the queue like a generator instead?

Reply:
@sanderegg I am not really happy with the solution I have here, but the point is the following: Ideally I would just do `yield await self._queue.get()` which will wait until a log enters the queue (https://docs.python.org/3/library/asyncio-queue.html#asyncio.Queue.get). I think that is the closest way to using a queue as a generator. But the point is that I cannot tell which view is the last one. So if the logs stop coming I need to check if  it is because the computation is done. Now, if the simulation runs for 3 weeks I would prefer if the api-server doesn't need to check if the computation is done every second. In fact I think if iSolve runs for a very long time it doesn't produce that many log lines. So what I do is I say if it lasted `n` seconds since the last log line arrived, then wait `max(n,30)` seconds before checking if a new log line has arrived. Do you have a cleaner solution for how to do this? Or do you think I simply shouldn't bother and just always wait 0.5 seconds here?